relayer/chains/cosmos/cosmos_chain_processor.go

```diff
@@ -523,14 +523,20 @@ func (ccp *CosmosChainProcessor) CurrentRelayerBalance(ctx context.Context) {
 			zap.Error(err),
 		)
 	}
-
+	address, err := ccp.chainProvider.Address()
+	if err != nil {
+		ccp.log.Error(
+			"Failed to get relayer bech32 wallet addresss",
+			zap.Error(err),
+		)
+	}
 	// Print the relevant gas prices
 	for _, gasDenom := range *ccp.parsedGasPrices {
 		for _, balance := range relayerWalletBalance {
 			if balance.Denom == gasDenom.Denom {
 				// Convert to a big float to get a float64 for metrics
 				f, _ := big.NewFloat(0.0).SetInt(balance.Amount.BigInt()).Float64()
-				ccp.metrics.SetWalletBalance(ccp.chainProvider.ChainId(), ccp.chainProvider.Key(), balance.Denom, f)
+				ccp.metrics.SetWalletBalance(ccp.chainProvider.ChainId(), ccp.chainProvider.Key(), address, balance.Denom, f)
```

Comment:
I wonder how we are using the same address in this loop. Is it supposed to be the same address for all iterations?

Reply:
Hmm, I'm not sure I full understand what you're saying. 
It should however be noted that this metric will NOT be logged if somehow the wallet balance is exactly 0.

Maybe we should consider removing/altering this `if` statement...

Reply:
I adjusted this logic. See note below.

Reply:
I'm unfamiliar with how the relayer works, so I was wondering how we use the same address for different denoms.